mParticle-Swrve/MPKitSwrve.m

```diff
@@ -0,0 +1,412 @@
+//     /*  Your code goes here.
+//         If the execution is not successful, please use a code other than MPKitReturnCodeSuccess for the execution status.
+//         Please see MPKitExecStatus.h for all exec status codes
+//      */
+//
+//     return [self execStatus:MPKitReturnCodeSuccess];
+//}
+
+/*
+    Implement this method if your SDK should be notified when user identities change
+*/
+//- (MPKitExecStatus *)onModifyComplete:(FilteredMParticleUser *)user request:(FilteredMPIdentityApiRequest *)request {
+//     /*  Your code goes here.
+//         If the execution is not successful, please use a code other than MPKitReturnCodeSuccess for the execution status.
+//         Please see MPKitExecStatus.h for all exec status codes
+//      */
+//
+//     return [self execStatus:MPKitReturnCodeSuccess];
+//}
+
+#pragma mark e-Commerce
+/*
+    Implement this method if your SDK supports commerce events.
+    If your SDK does support commerce event, but does not support all commerce event actions available in the mParticle SDK,
+    expand the received commerce event into regular events and log them accordingly (see sample code below)
+    Please see MPCommerceEvent.h > MPCommerceEventAction for complete list
+*/
+ - (MPKitExecStatus *)logCommerceEvent:(MPCommerceEvent *)commerceEvent {
+     MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:MPKitReturnCodeSuccess forwardCount:0];
+     NSString* currency = commerceEvent.currency ? : @"USD";
+
+     // In this example, this SDK only supports the 'Purchase' commerce event action
+     if (commerceEvent.action == MPCommerceEventActionPurchase) {
+             /* Your code goes here. */
+         
+         for (MPProduct *product in commerceEvent.products) {
+             SwrveIAPRewards* rewards = [[SwrveIAPRewards alloc] init];
+             [[SwrveSDK sharedInstance] unvalidatedIap:rewards localCost:[product.price doubleValue] localCurrency:currency productId:product.sku productIdQuantity:[product.quantity intValue]];
+             [execStatus incrementForwardCount];
+         }
+     } else { // Other commerce events are expanded and logged as regular events
+         NSArray *expandedInstructions = [commerceEvent expandedInstructions];
+
+         for (MPCommerceEventInstruction *commerceEventInstruction in expandedInstructions) {
+             [self logEvent:commerceEventInstruction.event];
+             [execStatus incrementForwardCount];
+         }
+     }
+
+     return execStatus;
+ }
+
+#pragma mark Events
+/*
+    Implement this method if your SDK logs user events.
+    Please see MPEvent.h
+*/
+ - (MPKitExecStatus *)logEvent:(MPEvent *)event {
+     /*  Your code goes here.
+         If the execution is not successful, please use a code other than MPKitReturnCodeSuccess for the execution status.
+         Please see MPKitExecStatus.h for all exec status codes
+      */
+     if (event.type == MPEventTypeOther) {
```

Comment:
are you sure you only want to accept events of `MPEventTypeOther`? most integrations support any custom event type, and it would be up the customer to filter which ones within the mParticle dashboard.

Reply:
Thanks for the comment @samdozor. This is simply capturing a special case where Swrve supports gifting of virtual currency via a specific API - we still capture all event types at the bottom of the function.

Reply:
ah - my bad!